hkmc2/shared/src/test/mlscript/ctx/TypeResolution.mls

```diff
@@ -0,0 +1,153 @@
+
+class Foo
+
+// * TODO: new-expression should also be resolvable for its type
+:rt
+new Foo
+//│ Resolved tree:
+//│ Blk:
+//│   stats = Nil
+//│   res = New:
+//│     cls = Ref{sym=member:Foo} of member:Foo
+//│     args = Nil
+//│     rft = N
+
+// * dynamic new-expression shouldn't be resolvable because of dynamic nature
```

Comment:
Dynamic doesn't mean we can _never_ resolvew; it means it's possible we can't in some cases.